src/libostree/ostree-sysroot-deploy.c

```diff
@@ -1361,6 +1362,57 @@ install_deployment_kernel (OstreeSysroot   *sysroot,
         }
     }
 
+
+  /* Copy other files that are stored in deployment's /usr/lib/ostree-boot. Lets keep this simple:
+   *
+   * - Ignore subdirectories. Only files in root of the /usr/lib/ostree-boot directory are copied.
+   *   There is an overhead of copying files to boot/, therefore the amount of files in a deployment's
+   *   usr/lib/ostree-boot should be kept to the minimum and subdirectories shouldn't really
+   *   be necessary.
+   *
+   * - Files in /boot are updated only with major releases, when kernel/initramfs
+   *   bootcsum changes. Files that require frequent updates should not be stored here.
+   */
+  if (!glnx_dirfd_iterator_init_take_fd (dup (tree_boot_dfd), &dfd_iter, error))
+    goto out;
+
+  while (TRUE)
+    {
+      struct dirent *dent;
+
+      if (!glnx_dirfd_iterator_next_dent (&dfd_iter, &dent, cancellable, error))
```

Comment:
FYI you can use `glnx_dirfd_iterator_next_dent_ensure_dtype` and look at `dent->d_type`, see e.g. https://github.com/ostreedev/ostree/blob/master/src/libostree/ostree-repo-refs.c#L136